packages/multi-select-combo-box/src/vaadin-multi-select-combo-box.js

```diff
@@ -823,6 +840,13 @@ class MultiSelectComboBox extends ResizeMixin(InputControlMixin(ThemableMixin(El
     this.dispatchEvent(new CustomEvent('change', { bubbles: true }));
   }
 
+  /** @private */
+  __updateTopGroup(groupSelectedItems, selectedItems, opened) {
+    if (groupSelectedItems && !opened) {
```

Comment:
I guess we should also clear `_topGroup` when the developer sets `groupSelectedItems` to `false`.

Reply:
The top group is not used in this case as `groupSelectedItems` is also checked by the internal combo-box.
However, I agree that it makes sense to clear it here. Updated code, also added unit tests for this case.